cylc/flow/templatevars.py

```diff
@@ -20,17 +20,19 @@
 from typing import Any, Dict
 
 from cylc.flow.exceptions import InputError
-
-
 from cylc.flow.rundb import CylcWorkflowDAO
+from cylc.flow.workflow_files import check_deprecation
 
 
 def get_template_vars_from_db(run_dir):
     """Get template vars stored in a workflow run database.
     """
     template_vars = {}
+    if check_deprecation(run_dir):
```

Comment:
I don't think back-compat mode is relevant here.

The Cylc database should be the same for Cylc 8 workflows whether in compat-mode or not.

We cannot restart Cylc 7 workflows at Cylc 8, `cylc play` protects against this, we need to ensure this check is performed earlier on for `cylc vr`.

Reply:
> The Cylc database should be the same for Cylc 8 workflows whether in compat-mode or not.

Yes, but what stops Cylc 8 `validate` being used on workflows run using Cylc 7?